Sources/Subs-Calendar.php

```diff
@@ -1851,4 +1851,21 @@ function removeHolidays($holiday_ids)
 	));
 }
 
+/**
+ * Helper function to convert date string to english
+ * so that date_parse can parse the date
+ *
+ * @param string $date A localized date string
+ * @return string English date string
+ */
+function convertDateToEnglish($date)
+{
+	global $txt, $context;
+
+	if ($context['user']['language'] == 'english')
+		return $date;
+
+	return str_replace(array_merge($txt['months_titles'], $txt['months_short']), array_merge(array(1 => 'January', 2 => 'February', 3 => 'March', 4 => 'April', 5 => 'May', 6 => 'June', 7 => 'July', 8 => 'August', 9 => 'September', 10 => 'October', 11 => 'November', 12 => 'December'), array(1 => 'Jan', 2 => 'Feb', 3 => 'Mar', 4 => 'Apr', 5 => 'May', 6 => 'Jun', 7 => 'Jul', 8 => 'Aug', 9 => 'Sep', 10 => 'Oct', 11 => 'Nov', 12 => 'Dec')), $date);
```

Comment:
```suggestion
	return strtr($date, array_combine(array_merge($txt['months_titles'], $txt['months_short']), array_merge(array('January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'), array('Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'))));
```
Using `strtr()` instead of `str_replace()` will ensure that double replacements never occur.

Reply:
Thanks for the review, updated